lib/help_scout/api/access_token.rb

```diff
@@ -1,10 +1,40 @@
 # frozen_string_literal: true
 
+require 'date'
+
 module HelpScout
   class API
     class AccessToken
       class << self
         def create
+          cache.present? ? fetch_token : request_token
+        end
+
+        def refresh!
```

Comment:
Feels like this method should maybe be a private method on `API::Client`. That's the only place it's used and there's already another private method for `#token_needs_refresh?`. Might pair up nicely…

```ruby
private

def token_needs_refresh?
  # …
end

def refresh_token
  HelpScout.api.access_token = HelpScout::API::AccessToken.create
end
```

Reply:
@jcypret I ended up going a different route and made `API::Client` less aware of token refresh details. Now, if `authorize?` is true, the token will refresh as needed (determined internally by the `API::AccessToken` class) and if found will be added as a Bearer auth header.